gusto/labels.py

```diff
@@ -30,14 +87,22 @@ def repl(t):
         Returns:
             :class:`Term`: the new term.
         """
-        test = t.form.arguments()[0]
-        new_form = ufl.replace(t.form, {test: new_test})
+        old_test = t.form.arguments()[0]
+        replace_dict = _replace_dict(old_test, new_test, idx, 'test')
+
+        try:
+            new_form = ufl.replace(t.form, replace_dict)
+        except Exception as err:
+            error_message = f"{type(err)} raised by ufl.replace when trying to" \
+                            + f" replace_test_function with {new_test}"
+            raise type(err)(error_message) from err
+
```

Comment:
Are we confident that we'll never have a perp on a test function?

Reply:
nope! but this would require some extra thought so I have made an issue: #364